webapp/src/components/Metrics/PerformanceAnalysis.test.tsx

```diff
@@ -0,0 +1,138 @@
+import { setupServer } from "msw/node";
+import {
+  getCustomMetricInfoAPIResponse,
+  getMetricsPerFilterAPIResponse,
+  getMetricsPerFilterAPIWithFailureResponse,
+} from "mocks/api/mockMetricsAPI";
+
+const setDatasetSplitName = jest.fn();
+const renderPerformanceAnalysis = (
+  availableDatasetSplits: AvailableDatasetSplits
+) =>
+  renderWithRouterAndRedux(
+    <PerformanceAnalysis
+      jobId="local"
+      pipeline={{ pipelineIndex: 0 }}
+      availableDatasetSplits={availableDatasetSplits}
+      datasetSplitName="eval"
+      setDatasetSplitName={setDatasetSplitName}
+    />,
+    { route: "/local?pipeline_index=0", path: "/:jobId?:pipeline" }
+  );
+describe("PerformanceAnalysis", () => {
+  const handlers = [
+    getCustomMetricInfoAPIResponse,
+    getMetricsPerFilterAPIResponse,
+  ];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  it("should have two toggle buttons for the train and test", () => {
+    renderPerformanceAnalysis({ train: true, eval: true });
+    // verify if the toggles displayed with correct names
+    expect(
+      screen.getByRole("button", { name: "Evaluation Set", pressed: true })
+    ).toHaveValue("eval");
+    expect(
+      screen.getByRole("button", { name: "Training Set", pressed: false })
+    ).toHaveValue("train");
+  });
+
+  it("should display the expected columns", async () => {
+    renderPerformanceAnalysis({ train: true, eval: true });
+    await waitFor(() => {
+      const expectedColumnHeaders = [
+        "Label",
+        "Total",
+        "Correct & Predicted",
+        "Correct & Rejected",
+        "Incorrect & Rejected",
+        "Incorrect & Predicted",
+        "Accuracy",
+        "Precision",
+        "Recall",
+        "F1",
+        "ECE",
+      ];
+      const actualColumnHeaders = screen.getAllByRole("columnheader");
+      expectedColumnHeaders.forEach((name, index) => {
+        name.includes("Correct")
+          ? expect(screen.getAllByTestId("CheckIcon")).toBeTruthy()
+          : name.includes("Incorrect")
+          ? expect(screen.getAllByTestId("XIcon")).toBeTruthy()
+          : expect(actualColumnHeaders[index].textContent).toEqual(name);
+      });
+    });
+  });
+
+  it("should display the default sort as desc for colum header 'Total number of utterances'", () => {
+    renderPerformanceAnalysis({ train: true, eval: true });
+    const header = screen.getByRole("columnheader", { name: /Total/ });
+    expect(header).toHaveAttribute("aria-sort", "descending");
+  });
+
+  it("should display the list of options if the user click on the column header 'filterValue'", () => {
+    renderPerformanceAnalysis({ train: true, eval: true });
+    fireEvent.mouseDown(screen.getByRole("button", { name: "Label" }));
+    const filterList = screen.getAllByRole("option");
+    expect(filterList).toHaveLength(10);
```

Comment:
If you move this down a bit, you'll be able to avoid hardcoding the value `10`.
```suggestion
    expect(filterList).toHaveLength(expectedList.length);
```